great_expectations/cli/datasource.py

```diff
@@ -147,23 +148,151 @@ def add_datasource(context):
     return data_source_name
 
 
-def profile_datasource(context, data_source_name, max_data_assets=20):
+def profile_datasource(context, data_source_name, data_assets=None, profile_all_data_assets=False, max_data_assets=20):
     """"Profile a named datasource using the specified context"""
+
+    msg_intro = """
+========== Profiling ==========
+
+Profiling '{0:s}' will create candidate expectations and documentation.
+
+Please note: Profiling is still a beta feature in Great Expectations.  The current profiler will evaluate the entire 
+data source (without sampling), which may be very time consuming. 
+As a rule of thumb, we recommend starting with data smaller than 100MB.
+
+To learn more about profiling, visit <blue>https://docs.greatexpectations.io/en/latest/guides/profiling.html\
+?utm_source=cli&utm_medium=init&utm_campaign={1:s}</blue>.
+"""
+
+    msg_confirm_ok_to_proceed = """
+Would you like to profile '{0:s}'?
+"""
+
+    msg_skipping = """Okay, skipping profiling for now. You can always do this
+later by running `great_expectations profile`.
+    """
+
+    msg_some_data_assets_not_found = """Some of the data assets you specified were not found: {0:s}    
+    """
+
+    msg_too_many_data_assets = """There are {0:d} data assets in {1:s}. Profiling all of them might take too long.    
+"""
+
+    msg_prompt_enter_data_asset_list = """Enter comma-separated list of data asset names (e.g., {0:s})   
+"""
+
+    msg_options = """Choose how to proceed:
+    1. Specify a list of the data assets to profile
+    2. Exit and profile later
+    3. Profile ALL data assets (this might take a while)
+"""
+
+    msg_data_doc_intro = """
+========== Data Documentation ==========
+
+To generate documentation from the data you just profiled, the profiling results should be moved from 
+great_expectations/uncommitted (ignored by git) to great_expectations/fixtures.
+
+Before committing, please make sure that this data does not contain sensitive information!
+
+To learn more: <blue>https://docs.greatexpectations.io/en/latest/guides/data_documentation.html\
+?utm_source=cli&utm_medium=init&utm_campaign={0:s}</blue>
+"""
+
+    cli_message(msg_intro.format(data_source_name, __version__.replace(".", "_")))
+
+    if data_assets:
+        data_assets = [item.strip() for item in data_assets.split(",")]
+
+    # Call the data context's profiling method to check if the arguments are valid
     profiling_results = context.profile_datasource(
         data_source_name,
-        max_data_assets=max_data_assets
+        data_assets=data_assets,
+        profile_all_data_assets=profile_all_data_assets,
+        max_data_assets=max_data_assets,
+        dry_run=True
```

Comment:
CLI lets the DataContext to check the arguments and decide if profiling can run. This is done by introducing the dry_run flag into DataContext.profile_datasource. When the flag is on, the method stops before running profiling and returns a result object that specifies whether profiling can run with the given arguments or some changes are required.